### PR TITLE
DDVSO593963 - Salon Iris AppCompat change documentation.

### PR DESCRIPTION
**Pull request by rladuca**

### CODE REVIEW

File: Documentation/compatibility/wpf - AppDomain shutdown handling may now invoke in cleanup of WeakEvents.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+## WPF AppDomain Shutdown Handling May Now Invoke in Cleanup of Weak Events
```
Comment (rpetrusha) [inline Documentation/compatibility/wpf - AppDomain shutdown handling may now invoke in cleanup of WeakEvents.md]: Invoke is an active verb; it requires a direct object -- the thing that is invoked. Do you mean to say "...may now invoke a Dispatcher from the Finalizer thread during Cleanup of Weak Events"?

File: Documentation/compatibility/wpf - AppDomain shutdown handling may now invoke in cleanup of WeakEvents.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+Edge
+
+### Version Introduced
+4.7.2
+
+### Source Analyzer Status
+NotPlanned
+
+### Change Description
+In .NET Framework 4.7.1 and earlier versions, WPF would potentially create a `System.Windows.Threading.Dispatcher` on the .NET finalizer thread during AppDomain shutdown.  This was fixed in
```
Comment (rpetrusha) [inline Documentation/compatibility/wpf - AppDomain shutdown handling may now invoke in cleanup of WeakEvents.md]: nit: would potentially create --> potentially creates
Dispatcher here and on line 15 can be a link: \<xref:System.Windows.Threading.Dispatcher?displayProperty=nameWithType>

File: Documentation/compatibility/wpf - AppDomain shutdown handling may now invoke in cleanup of WeakEvents.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+### Version Introduced
+4.7.2
+
+### Source Analyzer Status
+NotPlanned
+
+### Change Description
+In .NET Framework 4.7.1 and earlier versions, WPF would potentially create a `System.Windows.Threading.Dispatcher` on the .NET finalizer thread during AppDomain shutdown.  This was fixed in 
+.NET Framework 4.7.2 and later versions by making the cleanup of weak events thread-aware.  Due to this, the cleanup may invoke from the finalizer thread to the correct
```
Comment (rpetrusha) [inline Documentation/compatibility/wpf - AppDomain shutdown handling may now invoke in cleanup of WeakEvents.md]: there's the same problem here: something has to be invoked

File: Documentation/compatibility/wpf - AppDomain shutdown handling may now invoke in cleanup of WeakEvents.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+### Source Analyzer Status
+NotPlanned
+
+### Change Description
+In .NET Framework 4.7.1 and earlier versions, WPF would potentially create a `System.Windows.Threading.Dispatcher` on the .NET finalizer thread during AppDomain shutdown.  This was fixed in 
+.NET Framework 4.7.2 and later versions by making the cleanup of weak events thread-aware.  Due to this, the cleanup may invoke from the finalizer thread to the correct 
+`System.Windows.Threading.Dispatcher` in order to fully process cleanup.
+
+In certain applications, this change in finalizer timing can potentially cause exceptions during AppDomain or process shutdown.  This is generally seen in applications that do not correctly 
+shut down Dispatchers running on worker threads prior to process or AppDomain shutdown.  Such applications should take care to properly manage the lifetime of Dispatchers.
```
Comment (rpetrusha) [inline Documentation/compatibility/wpf - AppDomain shutdown handling may now invoke in cleanup of WeakEvents.md]: Dispatchers --> dispatchers (2 occurrences)

File: Documentation/compatibility/wpf - AppDomain shutdown handling may now invoke in cleanup of WeakEvents.md (hunk 1 above)
Comment (rladuca) [reply Documentation/compatibility/wpf - AppDomain shutdown handling may now invoke in cleanup of WeakEvents.md]: It should probably say "invoke on the associated Dispatcher" as invoking is a concept within WPF where we queue a delegate to a dispatcher for immediate or later execution.

File: Documentation/compatibility/wpf - AppDomain shutdown handling may now invoke in cleanup of WeakEvents.md (hunk 3 above)
Comment (rladuca) [reply Documentation/compatibility/wpf - AppDomain shutdown handling may now invoke in cleanup of WeakEvents.md]: I can clean this up as above.

File: Documentation/compatibility/wpf - AppDomain shutdown handling may now invoke in cleanup of WeakEvents.md (hunk 1 above)
Comment (rladuca) [reply Documentation/compatibility/wpf - AppDomain shutdown handling may now invoke in cleanup of WeakEvents.md]: Ended up going with "Calls Dispatcher.Invoke" (with appropriate tags) as its a bit more on the nose.

**Pull request merged**